### ISSUE TITLE
Fix warnings found by luacheck and enable luacheck on precommit

### ISSUE THREAD

**Issue by LeonidVas** [tarantool/tarantool#4681]
Now in repo exists broken lua code (and bad style code too). A good practice to use a linter before push something to a repo. I am suggest to create a config for the luacheck, add it to the repo and add a hook for a linter check on push to a nonpersonal branch.

**Comment by olegrok** [comment]
Related https://github.com/tarantool/tarantool/issues/4208

**Comment by ligurio** [comment]
Locations with Lua code in tarantool repo:

- test/
- src/box/
- src/lua/

**Comment by Gerold103** [comment]
`tarantoolctl.in` also is a Lua file https://github.com/tarantool/tarantool/blob/master/extra/dist/tarantoolctl.in

**Comment by ligurio** [comment]
v1 https://lists.tarantool.org/pipermail/tarantool-patches/2020-April/015872.html
v2 https://lists.tarantool.org/pipermail/tarantool-patches/2020-April/015556.html
v3 https://lists.tarantool.org/pipermail/tarantool-patches/2020-April/015375.html

**Comment by ligurio** [comment]
Blocked by:
- `test-run.py` https://github.com/tarantool/test-run/issues/208
- `tarantool-python` https://github.com/tarantool/tarantool-python/issues/157

**Comment by ligurio** [comment]
v4 https://lists.tarantool.org/pipermail/tarantool-patches/2020-April/016235.html

**Comment by ligurio** [comment]
TODO:

- add target `make lint`
- add parallelization with LuaLanes https://luarocks.github.io/luarocks/releases/luarocks-3.3.1.tar.gz

**Comment by ligurio** [comment]
```
lanes 3.13.0-0 is now installed in /builds/M4RrgQZ3/0/tarantool/tarantool/.rocks (license: MIT/X11)
.rocks/bin/luacheck --jobs  --codes --config .luacheckrc .
src/threading.c 828: pthread_create() failed, 1 EPERM
```

**Comment by ligurio** [comment]
v5 https://lists.tarantool.org/pipermail/tarantool-patches/2020-May/016648.html

**Comment by ligurio** [comment]
Splitted task for a number subtasks:

Fix luacheck warnings in test/app-tap #5453
Fix luacheck warnings in test/app #5454
Fix luacheck warnings in test/box #5455
Fix luacheck warnings in test/box-py #5456
Fix luacheck warnings in test/box-tap #5457
Fix luacheck warnings in test/engine #5458
Fix luacheck warnings in test/long_run-py #5460
Fix luacheck warnings in test/replication #5461
Fix luacheck warnings in test/replication-py #5462
Fix luacheck warnings in test/sql #5463
Fix luacheck warnings in test/sql-tap #5464
Fix luacheck warnings in test/vinyl #5466
Fix luacheck warnings in test/wal_off #5467
Fix luacheck warnings in test/xlog #5468
Fix luacheck warnings in test/xlog-py #5469
Add luacheck supression for luajit test #5470
